### ISSUE TITLE
TLS broken : vstest.console.exe still use .NET 4.5

### ISSUE THREAD

**Issue by tbolon** [microsoft/vstest#3733]
## Description

vstest.console project still target net4.5, beside the fact that it's not supported anymore since 2015.

It adds a problem: by default, .net apps build against .net version 4.5 [do not use TLS 1.2](https://docs.microsoft.com/en-us/dotnet/framework/network-programming/tls#for-net-framework-35---452-and-not-wcf), so if tests are using HTTP requests to call a server where only TLS 1.2+ is enabled (as per global recommendation) the test will fail.

My reproduction case happens when using Azure DevOps and using vstest task.

## Steps to reproduce

* Create a unit tests which requests an http resource only accessible through TLS 1.2
* Run the test using any test host (dotnet test, etc.), the test will pass
* Run the test using vstest.console.exe: the test will fail

```---- System.Net.WebException : The underlying connection was closed: An unexpected error occurred on a send.
-------- System.IO.IOException : Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.
------------ System.Net.Sockets.SocketException : An existing connection was forcibly closed by the remote host
```

## Expected behavior

The request should succeed using the most recent TLS protocol available

## Actual behavior

The request fail.

## Suggestion

You should at least update vstest.console.exe to 4.7.1+

**Comment by tbolon** [comment]
It seems that #3646 will fix this problem,

**Comment by nohwnd** [comment]
Still we will fix by moving to .net462 in next sprint.

**Comment by tbolon** [comment]
I was just coming back to suggest adding at least the correct AppContextSwitch `;Switch.System.Net.DontEnableSystemDefaultTlsVersions=false`. It seems you were quicker :)

**Comment by nloyolag** [comment]
Just wanted to jump into this thread to report that we are facing the same issue in our Azure DevOps build pipeline. Looking forward to the fix :)

**Comment by Evangelink** [comment]
Fixed by #3646
